chain/beacon/drand/drand.go

```diff
@@ -201,11 +204,32 @@ func (db *DrandBeacon) VerifyEntry(curr types.BeaconEntry, prev types.BeaconEntr
 	return err
 }
 
-func (db *DrandBeacon) MaxBeaconRoundForEpoch(filEpoch abi.ChainEpoch) uint64 {
+func (db *DrandBeacon) MaxBeaconRoundForEpoch(nv network.Version, filEpoch abi.ChainEpoch) uint64 {
 	// TODO: sometimes the genesis time for filecoin is zero and this goes negative
 	latestTs := ((uint64(filEpoch) * db.filRoundTime) + db.filGenTime) - db.filRoundTime
+
+	if nv <= network.Version15 {
+		return db.maxBeaconRoundV1(latestTs)
+	}
+
+	return db.maxBeaconRoundV2(latestTs)
+}
+
+func (db *DrandBeacon) maxBeaconRoundV1(latestTs uint64) uint64 {
 	dround := (latestTs - db.drandGenTime) / uint64(db.interval.Seconds())
 	return dround
 }
 
+func (db *DrandBeacon) maxBeaconRoundV2(latestTs uint64) uint64 {
+	if latestTs < db.drandGenTime {
+		return 1
+	}
+
+	fromGenesis := latestTs - db.drandGenTime
+	// we take the time from genesis divided by the periods in seconds, that
+	// gives us the number of periods since genesis.  We also add +1 because
+	// round 1 starts at genesis time.
+	return uint64(math.Floor(float64(fromGenesis)/db.interval.Seconds())) + 1
```

Comment:
I feel a bit uneasy using floats for this.
As you only need Floor it should be simple to do using pure ints.

Reply:
I agree, simplifying this.